### PR TITLE
Trie : Introduction and Implementation

### LINKED ISSUE #7228: Trie : Introduction and Implementation
### Description

Documentation and Implementation of User Defined Data Structure Trie along with code.

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by shreya024**
<hr>

## Description 📜

Added Documentation and Implementation of User Defined Data Structure Trie along with code.
Fixes #7228  

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: C_CPP/Data Structures/User Defined Data Structures/Trie/Trie.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<h1 align="center">  Trie or Prefix Tree </h1>
```
Comment (apeksha235) [inline C_CPP/Data Structures/User Defined Data Structures/Trie/Trie.md:1]: avoid using html tags for markdown formatting refer [this](https://www.markdownguide.org/basic-syntax/) instead

**Review by apeksha235 — CHANGES_REQUESTED** [review]
@shreya024

**Comment by shreya024** [comment]
@apeksha235 I have made the suggested change. Kindly check now.

**Review by apeksha235 — APPROVED** [review]
@shreya024 amazing!
@Saviour1001 please merge this

**Comment by shreya024** [comment]
@apeksha235 Thank you :)

**Pull request merged**